doctest/Main.hs

```diff
@@ -1,12 +1,31 @@
 module Main where
 
+import Data.Monoid ((<>))
+import System.FilePath ((</>))
+
+import qualified System.Directory
+import qualified System.IO.Temp
 import qualified Test.DocTest
 
 main :: IO ()
-main =
-    Test.DocTest.doctest
-        [ "-isrc"
-        , "src/Dhall.hs"
-        , "src/Dhall/Import.hs"
-        , "src/Dhall/Tutorial.hs"
-        ]
+main = do
+    pwd    <- System.Directory.getCurrentDirectory
+    prefix <- System.Directory.makeAbsolute pwd
+
+    System.IO.Temp.withSystemTempDirectory "doctest" $ \directory -> do
+        System.Directory.setCurrentDirectory directory
```

Comment:
This works.  Using `Test.Mockery.Directory.inTempDirectory` would be slightly more concise.

https://hackage.haskell.org/package/mockery-0.3.5/docs/Test-Mockery-Directory.html